MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication+Internal.swift

```diff
@@ -119,4 +119,21 @@ extension MSALNativeAuthPublicClientApplication {
         internalChallengeTypes.append(.redirect)
         return internalChallengeTypes
     }
+
+    static func convertChallengeTypes(
```

Comment:
Looking at these 2 functions, I think we took the paradigm from `MSALPublicClientApplication` but should they even be here? They don't use anything from the `MSALNativeAuthPublicClientApplication`, they could just as easily go into the Enums themselves like so:
![Screenshot 2024-06-04 at 15 36 34](https://github.com/AzureAD/microsoft-authentication-library-for-objc/assets/111577419/639bb4c2-8e82-4f28-876b-99189c6b6782)

Reply:
Given that these functions are now used by more than one class, they should go in a dedicated class (ex: challengeTypeConverter). I don't think that moving them to the MSALNativeAuthInternalChallengeType enum will make the situation better. I'd say to leave those functions in the publicClientApplication for now